src/array_interface.jl

```diff
@@ -16,53 +16,27 @@ ArrayInterfaceCore.indices_do_not_alias(::Type{ComponentArray{T,N,A,Axes}}) wher
 ArrayInterfaceCore.instances_do_not_alias(::Type{ComponentArray{T,N,A,Axes}}) where {T,N,A,Axes} = ArrayInterfaceCore.instances_do_not_alias(A)
 
 # Cats
-# TODO: Make this a little less copy-pastey
-function Base.hcat(x::AbstractComponentVecOrMat, y::AbstractComponentVecOrMat)
-    ax_x, ax_y = second_axis.((x,y))
-    if reduce((accum, key) -> accum || (key in keys(ax_x)), keys(ax_y); init=false) || getaxes(x)[1] != getaxes(y)[1]
-        return hcat(getdata(x), getdata(y))
+function Base.cat(inputs::ComponentArray...; dims::Int)
+    combined_data = cat(getdata.(inputs)...; dims=dims)
+    axes_to_merge = [(getaxes(i)..., FlatAxis())[dims] for i in inputs]
+    rest_axes = [getaxes(i)[1:end .!= dims] for i in inputs]
+    no_duplicate_keys = (length(inputs) == 1 || isempty(intersect(keys.(axes_to_merge)...)))
+    if no_duplicate_keys && length(Set(rest_axes)) == 1
```

Comment:
Note the extra check `length(Set(rest_axes)) == 1` that was not in place in the previous implementation.